packages/platforms/browser/lib/auto-instrumentation/network-request-plugin.ts

```diff
@@ -59,15 +69,49 @@ export class NetworkRequestPlugin implements Plugin<BrowserConfiguration> {
     span.setAttribute('http.method', startContext.method)
     span.setAttribute('http.url', networkRequestInfo.url)
 
-    return (endContext: RequestEndContext) => {
-      if (endContext.state === 'success') {
-        span.setAttribute('http.status_code', endContext.status)
-        this.spanFactory.endSpan(span, endContext.endTime)
-      }
+    return {
+      onRequestEnd: (endContext: RequestEndContext) => {
+        if (endContext.state === 'success') {
+          span.setAttribute('http.status_code', endContext.status)
+          this.spanFactory.endSpan(span, endContext.endTime)
+        }
+      },
+      extraRequestHeaders: this.getExtraRequestHeaders(
+        startContext,
+        span
```

Comment:
where a request is being tracked / network span created we use the supplied span